src/compiler/checker.ts

```diff
@@ -167,10 +167,10 @@ namespace ts {
 
         return checker;
 
-        function getEmitResolver(sourceFile?: SourceFile) {
+        function getEmitResolver(sourceFile: SourceFile, cancellationToken: CancellationToken) {

```

Comment:
My understanding was that we used a single point in the shim layer to have the LS-script side be able to communicate with a managed cancellation token.  But i was not under the impression that that meant that on the script side we not be passing the cancellation token along with an operation.  

IMO, it's a lot more complicated to use the 'host' pattern for cancellation.  With that pattern, anything that has access to the host may accidentally check for cancellation, even though they are not safe to cancel.  For example, any random type checking operation might check for this.  But it could end up dying.  This could lead the typechecker in a busted state that could corrupt all further calls into it.

For example, if we did things this way, then we'd have a problem for nearly all other typechecking operations.  For example, when we did quick info, we'd set have the cancellation token set in the host, then we'd call into the type checker, and it might throw a cancellation exception.  That cancellation exception would bubble out, leaving the type checker in an invalid state.

Right now though we only use cancellation for our expensive operation (i.e. getDiagnostics).  So even though the host has a cancellation token for pretty much every LS operation, the type checker is unaware of it except for the expensive call into it.  And that same expensive call is appropriately guarded for when cancellation may be thrown.